### PR TITLE
Purges all traces of that blasted library named Armadillo

### PR DESCRIPTION
**Pull request by KipHamiltons**
Armadillo and all traces are now removed.
We are building and passing tests. This step was surprisingly very easy.
Addresses all of the points in #406

### CODE REVIEW

**Comment by JosephusPaye** [comment]
I like how you've reacted to your own PR with hooray 😅

**Comment by KipHamiltons** [comment]
Couldn't be more excited 😀

File: shared/utility/math/geometry/Quad.hpp [hunk 1]
```diff
@@ -32,199 +31,6 @@ namespace math {
...
-            // template <typename U>
-            // friend std::ostream& operator<<(std::ostream& output, const std::vector<Quad<U>>& quads);
-        };
-
-        // *****************
-        // * EIGEN VERSION *
-        // *****************
-
         template <typename Scalar, int R, int C>
         class Quad<typename Eigen::Matrix<Scalar, R, C>> {
```
Comment (Bidski) [inline shared/utility/math/geometry/Quad.hpp:35]: We should be able to remove template specialisation from this class now
```suggestion
        class Quad {
```

**Review by Bidski — CHANGES_REQUESTED** [review]

File: shared/utility/math/geometry/Quad.hpp (hunk 1 above)
Comment (KipHamiltons) [reply shared/utility/math/geometry/Quad.hpp]: I removed it, but I don't really think it made it more readable in the end. I'll let you decide I think.

**Comment by KipHamiltons** [comment]
The auto-merge made a lot of mistakes because of the changed indentation and the huge amounts this PR deleted. This probably needs a proper review now after that :sob: 

P.S. can we like, never changed formatting (or header file extensions) like ever again lol... many hours of my life wasted to such changes ahhahaha

**Review by Bidski — APPROVED** [review]
LGTM

**Review by Bidski — APPROVED** [review]
Still LGTM

**Comment by JosephusPaye** [comment]
Nice one!

**Pull request merged**